frontend/components/puzzle.jsx

```diff
@@ -8,11 +8,14 @@ export default class Puzzle extends React.Component {
     socket.on('submit_response', (data) => {
       if (data.correct) {
         this.setState({ correct: data.correct });
-      }
+      } else {
+        this.setState({incorrectanswer: true});
+       }
     });
 
     this.state = {
       correct: false,
+      incorrectanswer: false
```

Comment:
let me know what you think, but i think we can simplify this to use a single variable by reusing `correct`:
1. initialize correct to `null` here - this indicates that no answer has been submitted before
2. remove the if statement around `this.setState({ correct: data.correct })` above
2. always pass it down to the child component: `<Component correct={correct}...`
3. in Puzzle2's AnswerAwareInput, distinguish between `false` and `null`: `${props.correct === false ? 'red' : 'black'}`